commands/system/ocr.swift

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/swift
+
+// Required parameters:
+// @raycast.schemaVersion 1
+// @raycast.title OCR
+// @raycast.mode silent
+
+// Optional parameters:
+// @raycast.icon 🖼
+// @raycast.packageName Productivity
+
+// Documentation:
+// @raycast.author zhe
+// @raycast.authorURL https://github.com/wmszhe
+// @raycast.description Use macOS Vision API Identification pictures, if it contain a QR code, Copy the QR code content to the clipboard, If do not include QR codes, identify text content and supplement to clipboard
+
+
+import Foundation
+import CoreImage
+import Cocoa
+import Vision
+
+let screenCapturePath = "/tmp/ocr.png"
+let recognitionLanguages = ["en-US", "zh-CN"]
+let joiner = " "
+
+func screenCapture(_ command: String) throws -> String{
+    let task = Process()
+    let pipe = Pipe()
+
+    task.launchPath = "/bin/bash"
+    task.arguments = ["-c", command]
+    task.standardOutput = pipe
+    task.standardError = pipe
+
+    try task.run()
+
+    let data = pipe.fileHandleForReading.readDataToEndOfFile()
+    return String(decoding: data, as: UTF8.self)
+}
+
+func convertCIImageToCGImage(inputImage: CIImage) -> CGImage? {
+    let context = CIContext(options: nil)
+    return context.createCGImage(inputImage, from: inputImage.extent)
+}
+
+func paste(text: String) {
+    let pasteboard = NSPasteboard.general
+    pasteboard.declareTypes([.string], owner: nil)
+    pasteboard.clearContents()
+    pasteboard.setString(text, forType: .string)
+}
+
+func recognizeTextHandler(request: VNRequest, error: Error?) {
+    guard let observations = request.results as? [VNRecognizedTextObservation] else {
+        return
+    }
+    let recognizedStrings = observations.compactMap { observation in
+        // Return the string of the top VNRecognizedText instance.
+        observation.topCandidates(1).first?.string
+    }
+
+    // Process the recognized strings.
+    let result = recognizedStrings.joined(separator: joiner)
+
+    print("The text content is: " + result)
+
+    paste(text: result)
+}
+
+func detectText(fileName: URL) {
+    if let ciImage = CIImage(contentsOf: fileName) {
+        guard let img = convertCIImageToCGImage(inputImage: ciImage) else {
+            return
+        }
+
+        let requestHandler = VNImageRequestHandler(cgImage: img)
+
+        // Create a new request to recognize text.
+        let request = VNRecognizeTextRequest(completionHandler: recognizeTextHandler)
+        request.recognitionLanguages = recognitionLanguages
+
+        do {
+            // Perform the text-recognition request.
+            try requestHandler.perform([request])
+        } catch {
+            print("Unable to perform the requests: \(error).")
+        }
+    }
```

Comment:
You are unwrapping the content from `CIImage(contentOf:)` in the line 72, why not do the same with this guard? Actually, instead of a `if let` you can move the guard up and instead of a new level, you have the whole method statement at the same level.

```suggestion
    guard 
        let ciImage = CIImage(contentsOf: fileName), 
        let img = convertCIImageToCGImage(inputImage: ciImage) 
    else {
        return
    }

    let requestHandler = VNImageRequestHandler(cgImage: img)

    // Create a new request to recognize text.
    let request = VNRecognizeTextRequest(completionHandler: recognizeTextHandler)
    request.recognitionLanguages = recognitionLanguages

    do {
        // Perform the text-recognition request.
        try requestHandler.perform([request])
    } catch {
        print("Unable to perform the requests: \(error).")
    }
```